src/query/procedure/callable_alias_mapper.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "callable_alias_mapper.hpp"
+
+#include <algorithm>
+#include <array>
+#include <filesystem>
+#include <fstream>
+
+#include <spdlog/spdlog.h>
+#include <json/json.hpp>
+
+#include "utils/logging.hpp"
+
+namespace memgraph::query::procedure {
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+CallableAliasMapper gCallableAliasMapper;
+
+void CallableAliasMapper::LoadMapping(const std::filesystem::path &path) {
+  using json = nlohmann::json;
+  if (!path.empty()) {
+    spdlog::info("Path to callable mappings was not set.");
+    return;
+  }
+
+  if (std::filesystem::exists(path)) {
+    const bool is_regular_file = std::filesystem::is_regular_file(path);
+    const bool has_json_extension = (path.extension() == ".json");
+    if (is_regular_file && has_json_extension) {
+      std::ifstream mapping_file(path);
+      try {
+        json mapping_data = json::parse(mapping_file);
+        mapping_ = mapping_data.get<std::unordered_map<std::string, std::string>>();
+      } catch (...) {
+        MG_ASSERT("Parsing callable mapping was unsuccesful. Make sure it is in correct json format.");
```

Comment:
What will `MG_ASSERT` do, will it also crash memgraph? How is that different than and why did you decide to use that instead of `spdlog::error`?

Reply:
spdlog::error will just print an error message, MG_ASSERT will close Memgraph because this is an unrecoverable error.
The user might not see the error log and think the everything is fine with the map, but it isn't.
MG_ASSERT is used in other places during startup when we cannot recover.

Reply:
Yes, forgot to add `false`. It should crash memgraph yes, if the user intended to specify the mapping file, but made an error with it, we should not wait just wait silently. Throwing would be another option, but other facilites that run upon startup tend to MG_ASSERT so I would just stick with that as well.